src/pymor/algorithms/basic.py

```diff
@@ -101,3 +102,22 @@ def project_array(U, basis, product=None, orthonormal=True):
         rhs = basis.inner(U, product)
         coeffs = scipy.linalg.solve(gramian, rhs, sym_pos=True, overwrite_a=True, overwrite_b=True).T
         return basis.lincomb(coeffs)
+
+
+def contains_zero_vector(vector_array, rtol=None, atol=None):
+    """returns `True` iff any vector in the array float_compares to 0s of the same dim
+
+    Parameters
+    ----------
+    vector_array
+        a |VectorArray| implementation
+    rtol
+        relative tolerance for float_cmp
+    atol
+        absolute tolerance for float_cmp
+    """
+    for i in range(len(vector_array)):
+        sup = vector_array[i].sup_norm()
+        if float_cmp(sup, 0.0, rtol, atol):
+            return True
+    return False
```

Comment:
```suggestion
    sup = np.max(vector_array.sup_norm())
    return float_cmp(sup, 0.0, rtol, atol):
```

Reply:
That's not actually equivalent though. Plus it always computes the norms for all vectors in the array.

Reply:
Sorry, should have been `min`. I have edited the suggestion. You are right about unnecessarily computing all norms. On the other hand, calling `sup_norm` on the entire array can be significantly faster (for `NumpyVectorArray`) when the result is false. I would prefer my solution, but if you clearly prefer your impl, we can leave it at that ..

Reply:
You are correct, for the `NumpyVectorArray` that would be faster. Those are only 1 in 6(7?) test executions though, currently. All others have per vector access to memory. SO I'll leave that for now.